snapshot/src/main/java/io/zeebe/snapshots/impl/SnapshotChecksum.java

```diff
@@ -72,9 +75,20 @@ private static long createCombinedChecksum(final List<Path> paths) throws IOExce
     final Checksum checksum = SnapshotChunkUtil.newChecksum();
 
     for (final var path : paths) {
-      if (!path.endsWith(CHECKSUM_FILE_NAME)) {
-        checksum.update(path.getFileName().toString().getBytes(StandardCharsets.UTF_8));
-        checksum.update(Files.readAllBytes(path));
+      final byte[] chunkId = path.getFileName().toString().getBytes(StandardCharsets.UTF_8);
+      checksum.update(chunkId);
+
+      try (final FileChannel channel = FileChannel.open(path, StandardOpenOption.READ)) {
+        final int bufferSize = 1024;
+        final ByteBuffer buff = ByteBuffer.allocate(bufferSize);
```

Comment:
We don't need to allocate this buffer for every bath - we can simply reuse it. I would also recommend using 4096 as the size, which is the typical page size/block size on most systems. See IoUtil.BLOCK_SIZE for example (or even reuse it :slightly_smiling_face:)

Reply:
> We don't need to allocate this buffer for every bath 

Oops. I don't know why I missed it :see_no_evil: 
Resuing IoUtil.BLOCK_SIZE is a good idea.